net/http/civetweb/civetweb.c

```diff
@@ -14308,6 +14308,10 @@ set_ssl_option(struct mg_context *ctx)
 #ifdef __clang__
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wincompatible-pointer-types"
+#endif
+#if defined(__GNUC__) || defined(__MINGW32__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wincompatible-pointer-types"
```

Comment:
I think that civetweb has some bugs with the use of incompatible pointer types that would be better to solve by using the right types rather than suppressing the warnings. For example, there are conversions from `void *` (64 bits) to `int` (32 bits), which lose half of the value.

```
/.../root/net/http/civetweb/civetweb.c(2577): warning #810: conversion from "void *" to "int" may lose significant bits
   		evhdl = (int)eventhdl;
```

Reply:
1. I guess, OPENSSL_VERSION_NUMBER one should use in source file - not in the cmake. And here already during compilation I need to specify OpenSSL version.

2. There are several hints why they have that warning. Here:
https://github.com/civetweb/civetweb/blob/master/src/civetweb.c#L15484-L15509
I right now submit two PRs to civetweb source.

Reply:
I still don't understand why `add_definitions(-DOPENSSL_API_1_1)` is needed.

As for the warnings, what I worry about it that the suppression will hide real bugs, but from the link you passed it seems it's quite confined, so it should be ok. However, I still think that the conversions from `void *` to `int` should be fixed. This just reminded me.

Reply:
> I still don't understand why add_definitions(-DOPENSSL_API_1_1) is needed.

It is how civetweb working.

They let compile source code with SSL support without linking ssl libraries directly.
And then loading library at the run time if required. And "fishing" all necessary symbols from loaded code. In principle, one could skip header files completely during compilation. 

OPENSSL_API_1_1 just inform expected SSL version by civetweb. 
I am not sure, that we can easily convince civetweb developers to change this logic. 
And with current civetweb 1.10 it is the only solution.

Reply:
Thank you for the explanation. I think this choice of civetweb devs is a bit unfortunate. I still think that `add_definitions()` is the wrong solution for us, because this will not be propagated to dependencies by CMake. Maybe that's not needed now? If `target_compile_definitions()` could be used, that would be better.

Reply:
> Maybe that's not needed now?

I think it is very rare case when openssl version changes between ROOT recompilations. 
But I add definitions with target_compile_definitions() if it helps to recognize such changes.

Reply:
That's not the problem. The problem is that the macro `OPENSSL_API_1_1` will only be defined during compilation of civetweb, and not during runtime or compilation of dependent targets. Then if something uses civetweb in interpreted code or other parts of ROOT, it may crash because it will be using the wrong interface. If no external code needs the macro to work properly, then it's fine.